core/src/surplus.ts

```diff
@@ -0,0 +1,88 @@
+import type { SurplusAuction, SurplusAuctionCollected, SurplusAuctionBaseData } from './types';
+import { getEarliestDate } from './helpers/getEarliestDate';
+import BigNumber from './bignumber';
+import getContract from './contracts';
+import { Contract } from 'ethers';
+import getNetworkDate from './date';
+import { RAD, WAD } from './constants/UNITS';
+
+const getSurplusAuctionLastIndex = async (contract: Contract): Promise<number> => {
+    const auctionsQuantityBinary = await contract.kicks();
+    return new BigNumber(auctionsQuantityBinary._hex).toNumber();
+};
+
+const getAuctionState = async (network: string, earliestEndDate: Date, greatestBid: number) => {
+    const isBidExpired = (await getNetworkDate(network)) > earliestEndDate;
+    const haveBids = greatestBid === 0;
+    if (haveBids) {
+        if (isBidExpired) {
+            return 'ready-for-collection';
+        }
+        return 'have-bids';
+    }
+    if (isBidExpired) {
+        return 'requires-restart';
+    }
+    return 'just-started';
+};
+
+export const fetchSurplusAuctionByIndex = async function (
+    network: string,
+    auctionIndex: number
+): Promise<SurplusAuction | SurplusAuctionCollected> {
```

Comment:
As suggested above, this function should return `SurplusAuction` which itself is a `SurplusAuctionActive | SurplusAuctionCollected`